generate/generate.go

```diff
@@ -653,3 +652,90 @@ func (ss *stringSet) slice() []string {
 	}
 	return res
 }
+
+func validateGeneratedFiles(generated []fileInfo) error {
+	logger := log.With().
+		Str("action", "generate.validateGeneratedFiles()").
+		Logger()
+
+	logger.Trace().Msg("validating generated files.")
+
+	genset := map[string]fileInfo{}
+	for _, file := range generated {
+		if other, ok := genset[file.Name()]; ok && file.Condition() {
+			return errors.E(ErrConflictingConfig,
+				"configs from %q and %q generate a file with same name %q have "+
+					"`condition = true`",
+				file.Origin(),
+				other.Origin(),
+				file.Name(),
+			)
+		}
+
+		if !file.Condition() {
+			continue
+		}
+
+		genset[file.Name()] = file
+	}
+
+	err := checkGeneratedFilesPaths(generated)
+	if err != nil {
+		return err
+	}
+
+	logger.Trace().Msg("generated files validated successfully.")
+	return nil
+}
+
+type dirGenFiles struct {
+	dir   string
+	files []string
+}
+
+// listGenFilesOutsideStacks returns a map of dir -> generated files.
+func listGenFilesOutsideStacks(rootdir, dir string) ([]dirGenFiles, error) {
```

Comment:
I think this could be much simpler if you handle both ListGenFiles() and the listGenFilesOutsideStacks() in the same function.
```
func SearchGeneratedFiles(rootdir string) (active []string, orphaned []string, err error) {
    err := filetype.WalkDir(rootdir, func(path string, d filetype.DirEntry, err error) error {
        if err != nil {
            return err
        }
        
        if d.IsDir() && filename.Base(path) == ".git" {
		    return filepath.SkipDir
		}
    
        if d.IsDir() {
            continue
        }
    
        data, err := os.ReadFile(path)
        // err handling
    
        if !hasGenHCLHeader(string(data)) {  
            return nil
        }
          
        if config.IsStack(filepath.Dir(path)) {
            active = append(active, path)
        } else {
            orphaned = append(orphaned, path)
        }

        return nil
    })
    
    if err != nil {
        return nil, nil, err
    }
    
    return active, orphaned, nil
}
```

Because in the end you need to traverse all files in the project anyway.
Would something like this be enough?

Reply:
Maybe, perhaps we could improve this in the future.